test/aqua/operators/test_pauli_sum_op.py

```diff
@@ -0,0 +1,189 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Test PauliSumOp """
+
+import unittest
+from test.aqua import QiskitAquaTestCase
+
+import numpy as np
+from scipy.sparse import csr_matrix
+
+from qiskit import QuantumCircuit, transpile
+from qiskit.aqua.operators import DictStateFn, I, PauliSumOp, SummedOp, X, Y, Z, Zero
+from qiskit.circuit import ParameterVector
+from qiskit.quantum_info import Pauli, SparsePauliOp
+
+
+class TestPauliSumOp(QiskitAquaTestCase):
+    """PauliSumOp tests."""
+
+    def test_construct(self):
+        """ constructor test """
+        sparse_pauli = SparsePauliOp(Pauli(label="XYZX"), coeffs=[2.0])
+        coeff = 3.0
+        pauli_sum = PauliSumOp(sparse_pauli, coeff=coeff)
+        self.assertIsInstance(pauli_sum, PauliSumOp)
+        self.assertEqual(pauli_sum.primitive, sparse_pauli)
+        self.assertEqual(pauli_sum.coeff, coeff)
+        self.assertEqual(pauli_sum.num_qubits, 4)
+
+    def test_add(self):
+        """ add test """
+        pauli_sum = 3 * X + Y
+        self.assertIsInstance(pauli_sum, PauliSumOp)
+
+        expected = PauliSumOp(
+            3.0 * SparsePauliOp(Pauli(label="X")) + SparsePauliOp(Pauli(label="Y"))
+        )
+
+        self.assertEqual(pauli_sum, expected)
+
+    def test_adjoint(self):
+        """ adjoint test """
+        pauli_sum = PauliSumOp(SparsePauliOp(Pauli(label="XYZX"), coeffs=[2]), coeff=3)
```

Comment:
Is there any good way to make a PualiSumOp object using Pauli class? `PauliSumOp(SparsePauliOp(Pauli(...), coeffs=[...]), coeff=...)` looks redundant. It maybe for next PR.

Reply:
I implemented `from_label()` method.
We can make this like `
PauliSumOp.from_list([("XYZX" , 2)], coeff=3)
`.

Reply:
from_label or from_list?

Reply:
sorry, from_list